kubechain/api/v1alpha1/taskruntoolcall_types.go

```diff
@@ -33,13 +45,16 @@ type TaskRunToolCallStatus struct {
 	Ready bool `json:"ready,omitempty"`
 
 	// Status indicates the current status of the tool call
-	// +kubebuilder:validation:Enum=Ready;Error;Pending;AwaitingHumanApproval;AwaitingHumanInput;AwaitingSubAgent
-	Status string `json:"status,omitempty"`
+	// +kubebuilder:validation:Enum=Ready;Error;Pending;AwaitingHumanApproval;AwaitingHumanInput;AwaitingSubAgent;ErrorRequestingHumanApproval
+	Status TaskRunToolCallStatusType `json:"status,omitempty"`
 
 	// StatusDetail provides additional details about the current status
 	// +optional
 	StatusDetail string `json:"statusDetail,omitempty"`
 
+	// HumanLayerCallId is the unique identifier for this function call in HumanLayer
+	HumanLayerCallId string `json:"humanLayerCallId"`
```

Comment:
thinking out loud - wondering if we want a kubechain-native ID, and either

1) use that as the call ID and don't call the field "HumanLayer"
2) support an "external call ID" as a first class thing in the humanlayer apis

but this is great for now, just voicing future thoughts